src/util/pointedthing.h

```diff
@@ -74,7 +74,12 @@ struct PointedThing
 	 * This is perpendicular to the face the ray hits,
 	 * points outside of the box and it's length is 1.
 	 */
-	v3s16 intersection_normal;
+	v3f intersection_normal;
+	/*!
+	 * Only valid if type isn't POINTEDTHING_NONE.
```

Comment:
```suggestion
	 * Only valid if type is POINTEDTHING_OBJECT.
```

This field is only set for object pointed things, right?

Reply:
No, nodes should have normals as well. The normals for nodes are more constrained though (always one of +-XYZ).

Reply:
Isn;t `raw_intersection_normal` only initialized for objects?

Reply:
oof yeah sorry, GH's preview showed this as `v3f intersection_normal; ... Only valid if type isn't POINTEDTHING_NONE.`